### PR TITLE
Cleanups in mean quadrature weights.

### PR DESCRIPTION
**Pull request by tarikdzanic**

### CODE REVIEW

File: pyfr/shapes.py [hunk 1]
```diff
@@ -162,6 +162,11 @@ def upts(self):
         rname = self.cfg.get(f'solver-elements-{self.name}', 'soln-pts')
         return get_quadrule(self.name, rname, self.nupts).pts
 
+    @cached_property
+    def upts_wts(self):
+        rname = self.cfg.get(f'solver-elements-{self.name}', 'soln-pts')
```
Comment (FreddieWitherden) [inline pyfr/shapes.py:167]: Can we use `_get_qrule` here to make it a one liner?

Comment (tarikdzanic) [reply pyfr/shapes.py:167]: I think `_get_qrule` would return the qpts weights if `quad-pts` or `quad-deg` are specified in the element section which isn't what we want

**Comment by FreddieWitherden** [comment]
I'm still not 100% here as some legitimate solution point sets do not have weights.  For example, the alpha optimised points.  Am I missing something?

**Comment by tarikdzanic** [comment]
I guess they do not have to be well-defined weights in the sense of they're all strictly positive, but all unique point sets should have a set of "weights" such that `w.u = int(u)` if `u` interpolates a polynomial on those point sets. This would just be the first row of the inverse Vandermonde matrix, which may have entries that are zero.

**Comment by FreddieWitherden** [comment]
> I guess they do not have to be well-defined weights in the sense of they're all strictly positive, but all unique point sets should have a set of "weights" such that `w.u = int(u)` if `u` interpolates a polynomial on those point sets. This would just be the first row of the inverse Vandermonde matrix, which may have entries that are zero.

Right, but the current rules in PyFR often do not have these weights in their tabulations.

**Comment by tarikdzanic** [comment]
Ah I see, in that case I'll close this and we can just use the weights from the inverse VDM matrix like before.

**Pull request closed without merging**